File: src/main/java/codereview/simpleorder/order/rest/ItemFeignClient.java
package codereview.simpleorder.order.rest;

import codereview.simpleorder.order.dto.ItemResponses;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

/**
 * TODO 테스트 하기 용이하게 변경하기
 * https://www.baeldung.com/spring-boot-running-port
 */

@FeignClient(value = "itemCall", url = "localhost:8080/items")
public interface ItemFeignClient {

    @GetMapping
    ItemResponses requestItems(@RequestParam("ids") List<Long> ids);
}

File: order/src/main/kotlin/msa/with/ddd/order/rest/ItemFeignClient.kt
package msa.with.ddd.order.rest

import msa.with.ddd.order.dto.web.ItemResponses
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam

/**
 * TODO 테스트 하기 용이하게 변경하기
 * https://www.baeldung.com/spring-boot-running-port
 */
@FeignClient(value = "item-service-call", url = "localhost:8000")
interface ItemFeignClient {

    @GetMapping("/item-service/items")
    fun requestItems(@RequestParam("ids") ids: List<Long>): ItemResponses
}